Adding DataLakeStore and DataLakeAnalytics services

vishrutshah opened the pull request.

veronicagg commented on Rakefile:
```diff
@@ -50,13 +57,21 @@ namespace :arm do
-  task :regen do
+  task :regen => :clean_generated do
     each_gem do |dir|
       execute_and_stream(OS.windows? ? 'del /S /Q lib\generated 2>nul' : 'rm -rf lib/generated')
```
should this line be removed from here?

vishrutshah replied:
yup not required anymore. 

---
In reply to: [70362516](https://github.com/vishrutshah/azure-sdk-ruby/pull/3#discussion_r70362516) [](ancestors = 70362516)

veronicagg commented on Rakefile:
```diff
@@ -219,5 +234,41 @@ REGEN_METADATA = {
+            spec_uri: 'https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-datalake-store/account/2015-10-01-preview/swagger/account.json',
+            ns: 'Azure::ARM::DataLakeStore::Account',
+            pn: 'azure_mgmt_datalake_store_account',
+            version: version,
```
should we include a comment on why only one contains version?

vishrutshah replied:
sure

---
In reply to: [70362869](https://github.com/vishrutshah/azure-sdk-ruby/pull/3#discussion_r70362869) [](ancestors = 70362869)

veronicagg commented on azure_mgmt_datalake_store/spec/spec_helper.rb:
```diff
@@ -0,0 +1,65 @@
+  def create_resource_group
+    resource_group_name = 'RubySDKTest_azure_mgmt_dl_store'
+    params = Azure::ARM::Resources::Models::ResourceGroup.new()
+    params.location = 'East US 2'
```
is there a reason for this to be 'East US 2'?

vishrutshah replied:
Yup because it's supported there. I didn't see option for East US on Azure portal UI.

---
In reply to: [70363355](https://github.com/vishrutshah/azure-sdk-ruby/pull/3#discussion_r70363355) [](ancestors = 70363355)

veronicagg commented on azure_arm.gemspec:
```diff
@@ -24,6 +24,8 @@ Gem::Specification.new do |spec|
   spec.add_runtime_dependency 'azure_mgmt_authorization', version
   spec.add_runtime_dependency 'azure_mgmt_cdn', version
   spec.add_runtime_dependency 'azure_mgmt_compute', version
+  spec.add_development_dependency 'azure_mgmt_datalake_analytics', version
```
>add_development_dependency [](start = 7, length = 26)

should this be a runtime_dependency?

vishrutshah replied:
yes good catch. 

---
In reply to: [70365105](https://github.com/vishrutshah/azure-sdk-ruby/pull/3#discussion_r70365105) [](ancestors = 70365105)

veronicagg commented:
@vishrutshah left some comments, rest looks good to me!

vishrutshah commented:
@veronicagg addressed the feedbacks. Please have a look at it once you get a chance. 

---
In reply to: [231913082](https://github.com/vishrutshah/azure-sdk-ruby/pull/3#issuecomment-231913082) [](ancestors = 231913082)

veronicagg commented:
@vishrutshah looks good, thanks!

vishrutshah commented:
@veronicagg Thanks for the review!!
